### ISSUE TITLE
loss values generated by model.fit and model.evaluate are not the same.

### ISSUE THREAD

**Issue by khlocs** [noelcodella/tripletloss-keras-tensorflow#4]
Thank you for providing such a valuable code for teaching tripletloss.
I read the code, generated .txt, and ran it successfully.
However, I got a strange problem.
When I feed "the training data" to do "mode.fit", the returned loss value say loss_1 = 0.4684. 
But, I feed exactly the same data into "model.evaluate" (right after the above code), I got loss_2 = 1.0.
The loss_1 should be equal (or similar) to loss_2, but It seems not like that.
I had tried for a couple of days to figure out what happened, but I still not found the answer.
It will be appreciated if you could provide some comments, Thanks a lot.
